services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -282,19 +350,39 @@ async def copy_file(
             copy_options |= dict(Callback=bytes_transfered_cb)
         await self.client.copy(**copy_options)
 
-    @s3_exception_handler(log)
+    @s3_exception_handler(_logger)
     async def list_files(
-        self, bucket: S3BucketName, *, prefix: str
+        self,
+        bucket: S3BucketName,
+        *,
+        prefix: str,
+        max_files_to_list: int = EXPAND_DIR_MAX_ITEM_COUNT,
     ) -> list[S3MetaData]:
-        # NOTE: adding a / at the end of a folder improves speed by several orders of magnitudes
-        response = await self.client.list_objects_v2(Bucket=bucket, Prefix=prefix)
+        """
+        NOTE: adding a / at the end of a folder improves speed by several orders of magnitudes
```

Comment:
Ok, so here I think I would have prefered that you just exposed the _list_objects above. but I guess it is ok for now, it is just weird that you put the logic of the max objects inside the client.